addons/website/static/src/components/dialog/edit_menu.js

```diff
@@ -122,40 +121,75 @@ export class EditMenuDialog extends Component {
                 { context: { lang: this.website.currentWebsite.metadata.lang } }
             );
             this.state.rootMenu = menu;
-            this.map = new Map();
             this.populate(this.map, this.state.rootMenu);
             this.toDelete = [];
         });
 
-        onMounted(() => {
-            this.$sortables = $(this.menuEditor.el);
-            this.$sortables.nestedSortable({
-                listType: 'ul',
-                handle: 'div',
-                items: 'li',
-                maxLevels: 2,
-                toleranceElement: '> div',
-                forcePlaceholderSize: true,
-                opacity: 0.6,
-                placeholder: 'oe_menu_placeholder',
-                tolerance: 'pointer',
-                attribute: 'data-menu-id',
-                expression: '()(.+)', // nestedSortable takes the second match of an expression (*sigh*)
-                isAllowed: (placeholder, placeholderParent, currentItem) => {
-                    return !placeholderParent
-                        || !currentItem[0].dataset.isMegaMenu && !placeholderParent[0].dataset.isMegaMenu;
-                },
-            });
+        useNestedSortable({
+            ref: this.menuEditor,
+            handle: "div",
+            nest: true,
+            maxLevels: 2,
+            onDrop: this._moveMenu.bind(this),
+            isAllowed: this._isAllowedMove.bind(this),
+            useElementSize: true,
         });
     }
 
+    get map() {
+        return this.state.flatMenus;
+    }
+
     populate(map, menu) {
         map.set(menu.fields['id'], menu);
         for (const submenu of menu.children) {
             this.populate(map, submenu);
         }
     }
 
+    _isAllowedMove(current, elementSelector) {
+        const currentIsMegaMenu = current.element.dataset.isMegaMenu === "true"
```

Comment:
Missing `;` (Unfortunately, the default eslint config is disabled on website, so this tends to happen)